unit-tests/func/frame-drops/test-frames-drops.py

```diff
@@ -0,0 +1,164 @@
+
+test.finish()
+
+#############################################################################################
+time.sleep(3) # allow some time for the sensor to closed before starting next test 
+#############################################################################################
+
+# Test #2
+
+test.start("Checking for frame drops when using sensor + syncer API")
+
+previous_depth_frame_number = -1
+previous_color_frame_number = -1
+
+syncer = rs.syncer()
+
+depth_sensor.open( high_fps_depth_profile )
+depth_sensor.start( syncer )
+
+color_sensor.open( low_fps_color_profile )
+color_sensor.start( syncer )
+
+frame = syncer.wait_for_frames()
+wait_frames_timer.start()
+
+while not wait_frames_timer.has_expired():
+    frame = syncer.wait_for_frames()
+    
+    depth_frame = frame.get_depth_frame()
+    color_frame = frame.get_color_frame()
+    
+    # The syncer can output only 1 frame (and not a frameset) if it wasn't synced
+    if depth_frame:
+        test.check_frame_drops(depth_frame, previous_depth_frame_number, 0, True)
+        previous_depth_frame_number = depth_frame.get_frame_number()
+
+    if color_frame:
+        test.check_frame_drops(color_frame, previous_color_frame_number, 0, True)
+        previous_color_frame_number = color_frame.get_frame_number()
+
+# Check that we received frames
+test.check(previous_depth_frame_number > 0)
+test.check(previous_color_frame_number > 0)
+
+depth_sensor.stop()
+depth_sensor.close()
+
+color_sensor.stop()
+color_sensor.close()
+
+test.finish()
+
+#############################################################################################
+time.sleep(3) # allow some time for the sensor to closed before starting next test 
+#############################################################################################
+
+# Test #3
+
+test.start("Checking for frame drops when using pipeline API")
+
+previous_depth_frame_number = -1
+previous_color_frame_number = -1
+
+pipeline = rs.pipeline()
+config = rs.config()
+
+config.enable_stream(rs.stream.depth, rs.format.z16, 30)
+config.enable_stream(rs.stream.color, rs.format.rgb8, 6)
+
+pipeline.start(config)
+
+frame = pipeline.wait_for_frames()
+
+wait_frames_timer.start()
+
+while not wait_frames_timer.has_expired():
+
+    frame = pipeline.wait_for_frames()
+    
+    # The pipeline always returns a frameset
+    test.check_frame_drops(frame.get_depth_frame(), previous_depth_frame_number, 0, True)
+    test.check_frame_drops(frame.get_color_frame(), previous_color_frame_number, 0, True)
+    
+    previous_depth_frame_number = frame.get_depth_frame().get_frame_number()
+    previous_color_frame_number = frame.get_color_frame().get_frame_number()
+
+# Check that we received frames
+test.check(previous_depth_frame_number > 0)
+test.check(previous_color_frame_number > 0)
+
+pipeline.stop()
+
+test.finish()
+#############################################################################################
+
+test.print_results_and_exit()
```

Comment:
Can we also show that there *are* drops if the syncer queue is 1? If it's 100% reproduction then it'll fail if something internal has changed (for the better).